zeek/compiler/glue-compiler.cc

```diff
@@ -124,9 +130,9 @@ static int extract_int(const std::string& chunk, size_t* i) {
     auto x = chunk.substr(*i, j - *i);
     *i = j;
 
-    int integer;
-    hilti::util::atoi_n(sint.begin(), sint.end(), 10, &integer);
-    return integer;
+    int integer = 0;
+    hilti::util::atoi_n(x.begin(), x.end(), 10, &integer);
```

Comment:
In the case where we got a sign, `i` is still pointing to that, right? And `atoi()` is examining the sign (at least the `-`) so aren't we now applying it twice?

Reply:
Good catch, fixed.